src/app/server/java/src/chip/appserver/ChipAppServer.java

```diff
@@ -17,11 +17,17 @@
  */
 package chip.appserver;
 
+import java.util.List;
+
 /** Controller to interact with the CHIP device. */
 public class ChipAppServer {
   private static final String TAG = ChipAppServer.class.getSimpleName();
 
   public native boolean startApp();
 
   public native boolean stopApp();
+
+  public native int getFabricCount();
```

Comment:
I would not provide two APIs to do almost the same thing...

- Most likely, if there are any Fabrics, the app will want to get the count and get the list immediately after (for display). Providing only one API for the list fits both needs. The performance hit when the count is 0 can probably be ignored.

- The risk of providing two APIs is that you may meet a race condition and get inconsistent results when calling both APIs sequentially, leading to inconsistent UI or other app problems.

Reply:
+1 - what is the purpose of getting the count separately from the list? If you have a list, the list has a `::size` already.

if there is a purpose, please describe it in some comment saying why this is useful as presumably a faster alternative for something.